mesonbuild/envconfig.py

```diff
@@ -403,6 +402,11 @@ def __init__(
                 if not isinstance(command, (list, str)):
                     raise mesonlib.MesonException(
                         f'Invalid type {command!r} for entry {name!r} in cross file')
+                if name == 'pkgconfig':
+                    if 'pkg-config' in binaries:
+                        raise mesonlib.MesonException('Both pkgconfig and pkg-config entries in machine file.')
+                    mlog.deprecation('"pkgconfig" entry is deprecated and should be replaced by "pkg-config"')
```

Comment:
This really only worked since 0.63.0 as far as I can tell, it would be nice to use FeatureDeprecated here... (IIRC @dcbaker particularly has had concerns about communicating the versions for things that arguably should never be done, and in this case people even have a very legitimate reason to do it...)